test/new-e2e/processes/docker_test.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/DataDog/datadog-api-client-go/v2/api/common"
+	"github.com/DataDog/datadog-api-client-go/v2/api/v2/datadog"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	containerName = "dd-agent"
+	agentImage    = "public.ecr.aws/datadog/agent:7"
+)
+
+type DockerTestSuite struct {
+	suite.Suite
+
+	ec2           *EC2TestEnv
+	ddAPIClient   *common.APIClient
+	procAPIClient *datadog.ProcessesApi
+}
+
+// In order for 'go test' to run this suite, we need to create
+// a normal test function and pass our suite to suite.Run
+func TestDockerTestSuite(t *testing.T) {
+	suite.Run(t, new(DockerTestSuite))
+}
+
+func (s *DockerTestSuite) SetupSuite() {
+	var err error
+
+	s.ec2, err = NewEC2TestEnv("process-agent-docker-test")
+	require.NoError(s.T(), err)
+
+	waitForDocker(s.T(), s.ec2.sshClient)
+
+	configuration := common.NewConfiguration()
+	s.ddAPIClient = common.NewAPIClient(configuration)
+	s.procAPIClient = datadog.NewProcessesApi(s.ddAPIClient)
+}
+
+func (s *DockerTestSuite) TearDownSuite() {
+	s.ec2.Close()
+}
+
+func (s *DockerTestSuite) TearDownTest() {
+	killAndRemoveContainers(s.T(), s.ec2.sshClient)
+}
+
+func (s *DockerTestSuite) TestProcessAgentOnDocker() {
+	hostName := createHostName(s.T().Name())
+	hostTag := fmt.Sprintf("host:%s", hostName)
+
+	s.T().Logf("start the datadog agent with hostname: %s", hostName)
+	// run the agent container on the VM
+	stdout, err := clients.ExecuteCommand(s.ec2.sshClient, fmt.Sprintf("sudo docker run -d --cgroupns host"+
+		" --name %s"+
+		" -v /var/run/docker.sock:/var/run/docker.sock:ro"+
+		" -v /proc/:/host/proc/:ro"+
+		" -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro"+
+		" -v /dd/config/:/etc/datadog-agent/"+
+		" -e DD_PROCESS_AGENT_ENABLED=true"+
+		" -e DD_HOSTNAME=%s"+
+		" -e DD_API_KEY=%s %s", containerName, hostName, s.ec2.ddAPIKey, agentImage))
+	s.T().Log(stdout)
+	require.NoError(s.T(), err)
+
+	s.waitForProcessAgent()
+
+	// Start containers
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name spring-hello -p 8080:8080 -d springio/gs-spring-boot-docker")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	stdout, err = clients.ExecuteCommand(s.ec2.sshClient,
+		"docker run --name sleepy  -d busybox sleep 9999")
+	s.T().Log(stdout)
+	assert.NoError(s.T(), err)
+
+	// Check processes are
+	ctx := common.NewDefaultContext(context.Background())
+	assert.Eventually(s.T(), func() bool {
+		resp, _, err := s.procAPIClient.ListProcesses(ctx, *datadog.NewListProcessesOptionalParameters().
+			WithSearch("java").
+			WithTags(fmt.Sprintf("%s,container_name:spring-hello", hostTag)).
+			WithPageLimit(2),
+		)
+		if err != nil {
+			s.T().Logf("Error when calling `ProcessesApi.ListProcesses`go: %v\n", err)
+			return false
+		}
+		if procs, ok := resp.GetDataOk(); ok && procs != nil {
+			return len(*procs) == 1
+		}
+		return false
+	}, 2*time.Minute, 1*time.Second)
```

Comment:
Might be worth extracting these consts so you could change them all at once in case we are falling out of the overall time limit for the test.